tests/test_updater_with_simulator.py

```diff
@@ -230,6 +232,32 @@ def test_snapshot_rollback_with_local_snapshot_hash_mismatch(self):
         with self.assertRaises(BadVersionNumberError):
             self._run_refresh()
 
+    @patch.object(builtins, "open", wraps=builtins.open)
+    def test_not_loading_targets_twice(self, wrapped_open: MagicMock):
+        # Do not load targets roles more than once when traversing
+        # the delegations tree
+
+        # Add new delegated targets, update the snapshot
+        spec_version = ".".join(SPECIFICATION_VERSION)
+        targets = Targets(1, spec_version, self.sim.safe_expiry, {}, None)
+        self.sim.add_delegation("targets", "role1", targets, False, ["*"], None)
+        self.sim.update_snapshot()
+
+        # Run refresh, top-level roles are loaded
+        updater = self._run_refresh()
+        # Clean up calls to open during refresh()
+        wrapped_open.reset_mock()
```

Comment:
you're adding a test for a specific bug fix, that makes sense... but this obviously leads to _"shouldn't we check that refresh opens correct files the correct number of times as well?"_ Maybe a follow up issue?

Reply:
I was planning to open an issue about testing the loading of metadata. Doing it now ...